src/DatumEither.ts

```diff
@@ -414,15 +426,138 @@ export const datumEither: Monad2<URI> & Traversable2<URI> & EitherM1<D.URI> = {
   reduceRight: reduceRightC,
 };
 
+// TODO: After we bump the min bound to >= 2.10, replace this with individual helper fns
+const eitherTDatum = getEitherM(D.Monad)
+
+/**
+ * @since 4.0.0
+ */
+export const Functor: Functor2<URI> = {
+  URI,
+  map: eitherTDatum.map
+}
+
+
+/**
+ * @since 4.0.0
+ */
+export const Bifunctor: Bifunctor2<URI> = {
+  URI,
+  mapLeft: eitherTDatum.mapLeft,
+  bimap: eitherTDatum.bimap
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Apply: Apply2<URI> = {
+  ...Functor,
+  ap: eitherTDatum.ap
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Chain: Chain2<URI> = {
+  ...Apply,
+  chain: eitherTDatum.chain
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Applicative: Applicative2<URI> = {
+  ...Apply,
+  of: eitherTDatum.of
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Alt: Alt2<URI> = {
+  ...Functor,
+  alt: eitherTDatum.alt
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Alternative: Alternative2<URI> = {
+  ...Alt,
+  ...Applicative,
+  zero: constInitial
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Monad: Monad2<URI> = {
+  ...Chain,
+  of: eitherTDatum.of
+}
+
+/**
+ * @since 4.0.0
+ */
+export const MonadThrow: MonadThrow2<URI> = {
+  ...Monad,
+  throwError: failure
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Foldable: Foldable2<URI> = {
+  URI,
+  reduce: reduceC,
+  reduceRight: reduceRightC,
+  foldMap: foldMapC
+}
+
+/**
+ * @since 4.0.0
+ */
+export const Traversable: Traversable2<URI> = {
+  ...Functor,
+  ...Foldable,
+  sequence: sequenceC,
+  traverse: traverseC
+}
+
+/**
+ * @since 4.0.0
+ */
+export const getProgressSemigroup = <E, A>(S: Semigroup<Either<E, A>>): Semigroup<DatumEither<E, A>> => 
```

Comment:
FYI These take either Semigroup instances because there's (at least) two reasonable Either Semigroups. This seemed better than doubling the DatumEither Semigroup/Monoid factory functions to account for those.

Reply:
Nice